client/src/components/markdown/RenkuMarkdownWithPathTranslation.tsx

```diff
@@ -28,30 +29,72 @@ const REF_TYPES = {
   FILE_PREV: "file", // with ! in the front
 };
 
-const getFileExtension = (file_path) => {
-  if (!file_path) return null;
+type IFileRef = {
+  data?: { content: string; file: File; file_name: string };
+  error: boolean;
+  filePreview?: boolean;
+  iBlock: number;
+  isOpened?: boolean;
+  refPath?: string;
+  type: string;
+};
 
-  if (file_path.match(/\.(.*)/) === null) return null;
-  return file_path.split(".").pop().toLowerCase();
+type RenkuMarkdownWithPathTranslationProps = {
+  /** The branch used to retrieve the file contents */
+  branch: string;
+  /** The GitLab client */
+  client: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  /** The file path to render */
+  filePath: string;
+  /** Is the file inside a project? */
+  insideProject: boolean;
+  /** The markdown text to display */
+  markdownText: string;
+  /** Sizes at which the UI shows a preview or not */
+  previewThreshold: FilePreviewProps["previewThreshold"];
+  /** The project id */
+  projectId: string;
+  /** The project path */
+  projectPathWithNamespace: string;
+  /** If true, render the output as a single line without line breaks */
+  singleLine?: boolean;
+  /** Any styles to apply */
+  style?: object;
 };
 
-const fileIsImage = (file_path) =>
-  IMAGE_EXTENSIONS.indexOf(getFileExtension(file_path)) >= 0;
+const getFileExtension = (file_path?: string) => {
+  if (file_path == null) return undefined;
+
+  if (file_path.match(/\.(.*)/) === null) return undefined;
+  const components = file_path.split(".");
+  if (components == null || components.length < 1) return undefined;
+  return components.pop()?.toLowerCase();
+};
+
+const fileIsImage = (file_path?: string) => {
+  if (file_path == null) return false;
+  const ext = getFileExtension(file_path);
+  if (ext == null) return false;
+  IMAGE_EXTENSIONS.indexOf(ext) >= 0;
```

Comment:
Use `.includes()` from es2016:

```suggestion
  const ext = getFileExtension(file_path);
  if (!ext) return false;
  return IMAGE_EXTENSIONS.includes(ext);
```